src/nomadic/summarize/compute.py

```diff
@@ -91,95 +106,50 @@ def calc_amplicons_summary(master_metadata, replicates_amplicon_qc_df):
     return samples_by_amplicons_summary_df
 
 
-variants_group_columns = [
-    "gene",
-    "chrom",
-    "pos",
-    # "ref",
-    # The alt might differ for multiallelic sites
-    # "alt",
-    "aa_change",
-    "aa_pos",
-    "mut_type",
-    "mutation",
-]
+def filter_false_positives(
+    variants_df: pd.DataFrame, min_obs: int = 1, min_wsaf: float = 0.15
+):
+    """Filter out likely false positive variant calls.
 
+    Remove variants that have only been observed `min_obs` times across all samples in
+    the analysis set, and with a WSAF below `min_wsaf`
 
-def filter_false_positives(variants_df: pd.DataFrame):
-    """Filter out likely false positive variant calls.
+    This removes likely false-positive mutations, which are usually rare and at low
+    WSAF.
 
-    Removes variants where only one mutation is called and the
-    maximum WSAF is below 15%.
     """
-    N_MUT = 1
-    WSAF_CUTOFF = 0.15
 
     mut = variants_df[variants_df["gt_int"].isin([1, 2])]
     df = variants_df.merge(
-        mut.groupby(variants_group_columns).agg(
+        mut.groupby(VARIANTS_GROUP_COLUMNS).agg(
             n_mut=pd.NamedAgg("gt_int", len), wsaf_max=pd.NamedAgg("wsaf", "max")
         ),
-        on=variants_group_columns,
+        on=VARIANTS_GROUP_COLUMNS,
     )
-    df = df[~(df["n_mut"].le(N_MUT) & df["wsaf_max"].lt(WSAF_CUTOFF))].drop(
+    df = df[~(df["n_mut"].le(min_obs) & df["wsaf_max"].lt(min_wsaf))].drop(
         columns=["n_mut", "wsaf_max"]
     )
     return df
 
 
-def compute_variant_prevalence(variants_df: pd.DataFrame) -> pd.DataFrame:
+def compute_variant_prevalence(
+    variants_df: pd.DataFrame,
+    master_df: pd.DataFrame = None,
+    additional_groups: list[str] = [],
+) -> pd.DataFrame:
     """
     Compute the prevalence of each mutation in `variants_df`
     """
 
-    prev_df = (
-        variants_df.groupby(
-            variants_group_columns,
+    if master_df is not None and additional_groups:
```

Comment:
I would also do this slighlty different. If you use additional groups, you probably wanted to filter, so I think it is better to error out if you than have no master df.